### PR TITLE
chore(website): expose examples on the website

### PR DESCRIPTION
**Pull request by jonaslagoni**
**Description**
This PR exposes (nearly) all the examples we have on the modelina website.

It works by reading all the directories in the examples folder, and each example has some Modelina code which is read from the `index.ts` file, and then the output code is read from the jest snapshots.

The overview in `/examples` is created from https://github.com/asyncapi/modelina/blob/master/examples/README.md with a few modifications such as changing links etc, but otherwise it's the same.

Here is what this feature enables:
- Show simple examples that contain modelina code and then the output.
- Each example link is shareable through the query parameter `/examples?selectedExample=adapting-input-and-output`
- Automatically updated on each website build (handles adding new examples, changes, etc.) with minimum maintenance as possible.

image

image

### CODE REVIEW

File: modelina-website/scripts/build-examples.js [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+  if(exampleName.includes('dart')) {
+    return 'dart';
+  }
+
+  if(exampleName.includes('go')) {
+    return 'go';
+  }
+
+  return 'typescript';
+}
```
Comment (magicmatatjahu) [inline modelina-website/scripts/build-examples.js:56]: Make some mapping 😆

**Review by magicmatatjahu — COMMENTED** [review]
Design/UX review:

- each links on the navigation should have `cursor-pointer` class to have "pointer" cursor icon when someone will hover it
- it will be nice if code and output will be rendered in this same line, not like current, which has some "margin" gap

  image

- readme should be rendered at center :)

  

I can accept but it will be good if you can propagate my suggestions :)

File: modelina-website/scripts/build-examples.js (hunk 1 above)
Comment (jonaslagoni) [reply modelina-website/scripts/build-examples.js:56]: Fine xD

**Comment by jonaslagoni** [comment]
>each links on the navigation should have cursor-pointer class to have "pointer" cursor icon when someone will hover it

Makes sense 👍 

> it will be nice if code and output will be rendered in this same line, not like current, which has some "margin" gap

Let me try 👍 

> readme should be rendered at center :)

Good point 😄

**Comment by jonaslagoni** [comment]
@magicmatatjahu adapted everything 🙂

**Review by magicmatatjahu — APPROVED** [review]
LGTM! :)

**Pull request merged**